vuepress/docs/next/docs/reference/bundle-upgrade.md

```diff
@@ -0,0 +1,136 @@
+
+The improvements include:
+* Simplified bundle directory structure
+* A decoupled build and deployment process that is easier to understand, modify, and troubleshoot
+* CI/CD integration and support 
+* One Git source repository required with Docker image repositories created automatically as needed
+* Optimized file storage and transfer methods using a layered file system, beneficial for large projects
+* Standardized bundle address format
+
+## Quick Reference
+* [Bundle Specifications](../curate/bundle-details.md)
+
+* [Bundle Management](../getting-started/ent-bundle.md)
+
+* [Bundle Version Comparison Table](./bundle-comparison.md)
+
+## Upgrade Process
+
+### Initialize Bundle 
+
+1. Initialize a new bundle to set up the standard Entando Bundle structure using the ent CLI:
+
+``` 
+ent bundle init YOUR-BUNDLE-NAME
+```
+### Copy Assets
+1. Use the ent CLI bundler command to export assets from an existing Entando Bundle. See the step-by-step [Export and Publish Tutorial](../../tutorials/create/pb/export-bundle-from-application.md). 
+
+2. `YOUR-BUNDLE-NAME/platform` directory is designated for resources such as fragments, contents, pages, and templates. Copy the output of the ent bundler to this folder. You can also accomplish this manually but there are advantages to the bundler such as generating top level descriptors for platform resources.
+Note that widgets that are not MFEs are considered platform entities on Entando and should be placed in the `platform/widgets` directory.
+
+### Add Micro Frontends & Microservices
+ 
+1. Add your micro frontend to the new bundle descriptor `entando.json`:
+``` sh
+ent bundle mfe add YOUR-MFE1
+```
+The docker-based bundle has a centralized bundle descriptor with bundle entities such as micro frontends, microservices, APIs, DBMS, and ID management specifications. 
+
+2. Copy the MFE files into `YOUR-BUNDLE-NAME/microfrontends` directory. For instance, for an MFE named YOUR-MFE1: 
+
+``` sh
+mv YOUR-ORIGIN-DIRECTORY/YOUR-MFE1/{.,}* microfrontends/YOUR-MFE1
+```
+Micro frontends and microservices are partitioned into their own directories in the docker-based bundles to streamline the CI/CD process.
+
+3. Add your microservice:
+``` sh
+ent bundle ms add YOUR-MS1
+```
+4. Copy the microservice YOUR-MS1 files to the new location:
+``` sh
+mv YOUR-ORIGIN-DIRECTORY/YOUR-MS1/{.,}* microservicess/YOUR-MS1
+```
+5. Repeat steps 1 through 4 for all micro frontends and microservices.
+
+### Add API Claims
+1. Now add an API claim to connect the `YOUR-MFE1` to a microsersive as needed. The connection information is added to the descriptor `entando.json`.
+```shell
+ent bundle api add YOUR-MFE1 YOUR-API --serviceName=YOUR-MS1 --serviceUrl=http://localhost:8081
+```
+
+2. Repeat the previous step for all APIs as needed.
+
+### Add Custom Commands to Test Locally
+
+1. For local development and testing, a custom command is needed in the `entando.json` to use a different port for each MFE as shown below: 
```

Comment:
**In** local dev, not **for**

sentence can be simplified "**In** local development and testing, the `entando.json` requires a custom command for each MFE to use a different port:"

Reply:
i see the distinction. i think this is a specific case - the command is necessary while you are testing locally. both work; i was cueing off of the immediacy and action of putting the custom command into effect (vs offering it as an alternative, which is distance and optional). same coin, 2 sides :)

"needed in the" is equivalent to "required by" here, although the former doesn't assign a requiring entity and the latter assigns the json as the requiring entity. using different ports requires something from the json, which in turn requires a modification. however, the requirement is indirectly imposed by the configuration. it's probably best to say "To assign each MFE a different port in local development, the `entando.json` must include the following custom command" etc.